File: python27-app/app_identity.py
import webapp2
import json
from google.appengine.api import app_identity
import base64


class ProjectIDHandler(webapp2.RequestHandler):
  def get(self):
    self.response.write(app_identity.get_application_id())


class HostnameHandler(webapp2.RequestHandler):
  def get(self):
    self.response.write(app_identity.get_default_version_hostname())


class AccessTokenHandler(webapp2.RequestHandler):
  def get(self):
    auth_token, expiration_time = app_identity.get_access_token(
        'https://www.googleapis.com/auth/cloud-platform')
    json.dump({'token': auth_token, 'expires': expiration_time}, self.response)


class SignBlobHandler(webapp2.RequestHandler):
  def get(self):
    encoded_blob = self.request.get('blob').encode('utf-8')
    blob = base64.urlsafe_b64decode(encoded_blob)
    key_name, signature = app_identity.sign_blob(blob)
    json.dump({'key_name': key_name, 'signature': base64.b64encode(signature)},
              self.response)


class CertificateHandler(webapp2.RequestHandler):
  def get(self):
    certs = []
    for cert in app_identity.get_public_certificates():
      certs.append({'key_name': cert.key_name,
                    'pem': cert.x509_certificate_pem})

    json.dump(certs, self.response)


class ServiceAccountNameHandler(webapp2.RequestHandler):
  def get(self):
    self.response.write(app_identity.get_service_account_name())


urls = [
    ('/python/app_identity/project_id', ProjectIDHandler),
    ('/python/app_identity/hostname', HostnameHandler),
    ('/python/app_identity/access_token', AccessTokenHandler),
    ('/python/app_identity/sign', SignBlobHandler),
    ('/python/app_identity/certificates', CertificateHandler),
    ('/python/app_identity/service_account_name', ServiceAccountNameHandler)
]

File: php-app/app_identity.php
<?php

require_once 'google/appengine/api/app_identity/AppIdentityService.php';

use google\appengine\api\app_identity\AppIdentityService;

function projectIDHandler() {
    echo AppIdentityService::getApplicationId();
}

function hostnameHandler() {
    echo AppIdentityService::getDefaultVersionHostname();
}

function accessTokenHandler() {
    $access_token = AppIdentityService::getAccessToken(
        'https://www.googleapis.com/auth/cloud-platform');
    $output = array('token' => $access_token['access_token'],
                    'expires' => $access_token['expiration_time']);
    echo json_encode($output);
}

function signBlobHandler() {
    $encoded_blob = strtr($_GET['blob'], '-_', '+/'); // Convert urlsafe_base64 chars.
    $blob = base64_decode($encoded_blob);
    $signature = AppIdentityService::signForApp($blob);
    $signature['signature'] = base64_encode($signature['signature']);
    echo json_encode($signature);
}

function certificateHandler() {
    $certs = array();
    foreach(AppIdentityService::getPublicCertificates() as $certificate) {
        $certs[] = array('key_name' => $certificate->getCertificateName(),
                         'pem' => $certificate->getX509CertificateInPemFormat());
    }
    echo json_encode($certs);
}

function serviceAccountNameHandler() {
    echo AppIdentityService::getServiceAccountName();
}
